docs/generate_docs.py

```diff
@@ -0,0 +1,108 @@
+#!/usr/bin/env python3
+
+import shutil
+import subprocess
+from pathlib import Path
+
+
+def run_command(command: list[str]) -> None:
+    """Run a command and exit if it fails."""
+    print(f"Running: {' '.join(command)}")
+    try:
+        subprocess.run(command, check=True)
+    except subprocess.CalledProcessError as e:
+        print(f"Warning: Command failed: {' '.join(command)}")
+        print(f"Error: {e}")
+
+
+def main():
+    docs_dir = Path(__file__).parent
+    project_root = docs_dir.parent
+
+    # Install documentation requirements
+    requirements_file = docs_dir / "requirements.txt"
+    run_command(["pip", "install", "-r", str(requirements_file)])
+
+    # Install core package
+    run_command(["pip", "install", "-e", "."])
+
+    # Install all service dependencies
+    services = [
+        # Core services
+        "openai",
+        "langchain",
+        # Audio/Speech services
+        "whisper",
+        "noisereduce",
+        "soundfile",
+        # Cloud providers
+        "aws",
+        "azure",
+        "google",
+        # AI services
+        "anthropic",
+        "assemblyai",
+        "deepgram",
+        "elevenlabs",
+        "fal",
+        "fireworks",
+        "gladia",
+        "grok",
+        "groq",
+        "lmnt",
+        "moondream",
+        "nim",
+        "openpipe",
+        "playht",
+        # Transport services
+        "websocket",
+        "daily",
+        "livekit",
+        # Additional services
+        "silero",
+        "cartesia",
+        "canonical",
+    ]
+
+    extras = ",".join(services)
+    try:
+        run_command(["pip", "install", "-e", f".[{extras}]"])
+    except Exception as e:
+        print(f"Warning: Some dependencies failed to install: {e}")
+
+    # Clean old files
+    api_dir = docs_dir / "api"
+    build_dir = docs_dir / "_build"
+    for dir in [api_dir, build_dir]:
+        if dir.exists():
+            shutil.rmtree(dir)
+
+    # Generate API documentation
+    run_command(
+        [
+            "sphinx-apidoc",
+            "-f",  # Force overwrite
+            "-e",  # Put each module on its own page
+            "-M",  # Put module documentation before submodule
+            "--no-toc",  # Don't generate modules.rst (cleaner structure)
+            "-o",
+            str(api_dir),  # Output directory
+            str(project_root / "src/pipecat"),
+            # Exclude problematic files and directories
+            "**/processors/gstreamer/*",  # Optional gstreamer
```

Comment:
These were giving me trouble. I need to revisit to see if I can avoid conflicts.

Do we need/want these?